### PR TITLE
add TYLinkLabel

### PR DESCRIPTION
**Pull request by selinersev**

### CODE REVIEW

**Review by batuhansk — APPROVED** [review]

**Comment by batuhansk** [comment]
👏🏼👏🏼

**Review by unalCe — APPROVED** [review]

**Review by aniltaskiran — APPROVED** [review]

**Review by AhmettKeskin — APPROVED** [review]

File: UILibraries/LinkLabel/LinkLabel/Classes/LinkLabel.swift [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+        for link in links where isTapped(link: link, tapIndex: tapIndex){
+            didTap(for: link)
+            return
+        }
+    }
+
+    private func isTapped(link:LabelLink, tapIndex:Int) -> Bool{
+        let linkedText = link.title
+        guard let linkedTextRange = attributedText?.string.range(of: linkedText),
+              let targetRange = attributedText?.string.range(from: linkedTextRange) else {return false}
```
Comment (mustafagunes) [inline UILibraries/LinkLabel/LinkLabel/Classes/LinkLabel.swift]:
```suggestion
              let targetRange = attributedText?.string.range(from: linkedTextRange) else { return false }
```
😅

File: UILibraries/LinkLabel/LinkLabel/Classes/LinkLabel.swift [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+      textContainer.lineFragmentPadding = 0
+      textContainer.maximumNumberOfLines = numberOfLines
+      textContainer.lineBreakMode = lineBreakMode
+      layoutManager.addTextContainer(textContainer)
+
+      let index = layoutManager.characterIndex(for: point,
+                                               in: textContainer,
+                                               fractionOfDistanceBetweenInsertionPoints: nil)
+      return index
+    }
```
Comment (mustafagunes) [inline UILibraries/LinkLabel/LinkLabel/Classes/LinkLabel.swift]: Indentation must be 4 spaces 👍🏻

File: UILibraries/LinkLabel/LinkLabel/Classes/LinkLabel.swift [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+            }
+        }
+        self.attributedText = attributedText
+        customize = false
+    }
+
+    @objc private func tapLabel(gesture: UITapGestureRecognizer) {
+        let tapLocation = gesture.location(in: self)
+        let tapIndex = indexOfAttributedTextCharacter(at: tapLocation)
+        for link in links where isTapped(link: link, tapIndex: tapIndex){
```
Comment (mustafagunes) [inline UILibraries/LinkLabel/LinkLabel/Classes/LinkLabel.swift]:
```suggestion
        for link in links where isTapped(link: link, tapIndex: tapIndex) {
```

**Review by mustafagunes — COMMENTED** [review]
I have a few simple suggestions. Good work ✋

**Pull request merged**